karaf/apache-brooklyn/src/main/resources/etc/org.ops4j.pax.logging.cfg

```diff
@@ -18,29 +18,74 @@
-log4j.appender.out.file=${karaf.home}/log/brooklyn.log
+log4j.appender.out.layout.ConversionPattern=%d{ABSOLUTE} %-5.5p %3.3X{bundle.id} %-30.30c{2} [%-16.16t] %m%n
+log4j.appender.out.file=${karaf.home}/log/brooklyn.info.log
+log4j.appender.out.Threshold=INFO
 log4j.appender.out.append=true
 log4j.appender.out.maxFileSize=1MB
 log4j.appender.out.maxBackupIndex=10
 
+# Debug file appender
+log4j.appender.debugFile=org.apache.log4j.RollingFileAppender
+log4j.appender.debugFile.layout=org.apache.log4j.PatternLayout
+log4j.appender.debugFile.layout.ConversionPattern=%d{ABSOLUTE} %-5.5p %3.3X{bundle.id} %-30.30c{2} [%-16.16t] %m%n
+log4j.appender.debugFile.file=${karaf.home}/log/brooklyn.debug.log
+log4j.appender.debugFile.append=true
+log4j.appender.debugFile.maxFileSize=1MB
+log4j.appender.debugFile.maxBackupIndex=10
+
 # Sift appender
 log4j.appender.sift=org.apache.log4j.sift.MDCSiftingAppender
 log4j.appender.sift.key=bundle.name
 log4j.appender.sift.default=brooklyn
 log4j.appender.sift.appender=org.apache.log4j.FileAppender
 log4j.appender.sift.appender.layout=org.apache.log4j.PatternLayout
-log4j.appender.sift.appender.layout.ConversionPattern=%d{ABSOLUTE} | %-5.5p | %-16.16t | %-32.32c{1} | %-32.32C %4L | %m%n
+log4j.appender.sift.appender.layout.ConversionPattern=%d{ABSOLUTE} %-5.5p %3.3X{bundle.id} %-30.30c{2} [%-16.16t] %m%n
 log4j.appender.sift.appender.file=${karaf.data}/log/$\\{bundle.name\\}.log
-log4j.appender.sift.appender.append=true
\ No newline at end of file
+log4j.appender.sift.appender.append=true
+
+
+# Logger configuration
+
+# The following properties turn on quite verbose DEBUG logging for Brooklyn-relevant loggers
+log4j.logger.brooklyn=DEBUG
+log4j.logger.org.apache.brooklyn=DEBUG
+log4j.logger.org.jclouds=DEBUG
+log4j.logger.jclouds=DEBUG
+
+# If you're just going to have a few debug categories these are recommended;
+# comment out the loggers above and leave these as-is
+log4.logger.brooklyn.SSH=DEBUG
+log4.logger.brooklyn.location.basic.jclouds=DEBUG
+log4.logger.brooklyn.util.internal.ssh=DEBUG
+log4.logger.org.apache.brooklyn.SSH=DEBUG
+log4.logger.org.apache.brooklyn.location.basic.jclouds=DEBUG
+log4.logger.org.apache.brooklyn.util.internal.ssh=DEBUG
+
+
+# some loggers are very noisy however, exclude them, or make them file only
+log4j.logger.org.apache.cxf=ERROR
+log4j.logger.net.schmizz=WARN
+log4j.logger.org.eclipse.jetty=WARN
+log4j.logger.org.reflections.Reflections=WARN
+log4j.logger.com.sun.jersey.server.impl.application=WARN
+log4j.logger.org.apache.whirr.service.ComputeCache=WARN
+log4j.logger.jclouds.ssh=WARN
+log4j.logger.org.apache.http.impl.client=WARN
+log4j.logger.javax.management.remote=WARN
+log4j.logger.com.wordnik.swagger=WARN
+log4j.logger.org.mongodb.driver=WARN
+log4j.logger.org.apache.brooklyn.location.jclouds.JcloudsPropertiesFromBrooklynProperties=WARN
```

Comment:
When I started Brooklyn up and pointed a web server at it, but didn't actually do anything, I got bunches of log lines from that class every few seconds, not really telling me anything but filling up the logs, e.g.

JcloudsPropertiesFromBrooklynProperties.getProviderName NOT changing provider from openstack-nova to candidate bluebox

This setting will avoid the logs filling up with these messages.